.github/workflows/cross-version-tests.yml

```diff
@@ -30,7 +30,7 @@ on:
     - cron: "0 13 * * *"
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
```

Comment:
`github.ref` is:

> The fully-formed ref of the branch or tag that triggered the workflow run. For workflows triggered by push, this is the branch or tag ref that was pushed. For workflows triggered by pull_request, this is the pull request merge branch. For workflows triggered by release, this is the release tag created. For other triggers, this is the branch or tag ref that triggered the workflow run. This is only set if a branch or tag is available for the event type. The ref given is fully-formed, meaning that for branches the format is refs/heads/<branch_name>, for pull requests it is refs/pull/<pr_number>/merge, and for tags it is refs/tags/<tag_name>. For example, refs/heads/feature-branch-1.
--

source: https://docs.github.com/en/actions/learn-github-actions/contexts

Reply:
On pr, `github.head_ref` is used, on push `github.ref` is used.

Reply:
The workflow runs triggered on the master branch belong to the same group. Only the latest one is executed, the rest is cancelled.

Reply:
This isn't ideal, but better than waiting 3 or 4 hours (which significantly slows down iteration cycle) until CI starts running on your PR.